core/common/src/main/java/alluxio/conf/Deprecated.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.conf;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+/**
+ * An annotation for PropertyKeys that can be used to mark a key as deprecated (slated for
+ * removal). If a key is marked with this annotation, then any Alluxio processes should trigger
+ * log warning messages upon configuration validation.
+ *
+ * This annotation is provided as an alternative to {@link java.lang.Deprecated}. Both
+ * annotations do not need to be specified on a key. The advantage of using this annotation is
+ * that we can supply a custom message rather than simply telling the user "This key has been
+ * deprecated".
+ */
+@Retention(RetentionPolicy.RUNTIME)
+public @interface Deprecated {
+  /**
+   * A message to the user which could suggest an alternative key.
+   *
+   * The message should not mention anything about deprecation, but rather alternatives,
```

Comment:
why shouldn't the message mention anything about deprecation?

Reply:
If the key is marked with `@Deprecated` we don't need a sentence in the message that says "This key has been deprecated". The log warning message will do it for us. If the programmer writes something like that in the message, it's redundant. I.e. if the log message has the format

```
String.format("%s is deprecated and will be removed in the future. %s", keyName, annotationMessage)
```

Then the logged message might read which doesn't sound right:

```
alluxio.x.y.z is deprecated and will be removed in the future. This key has been deprecated...Please use x.y.z instead
```